third_party/amd/lib/TritonAMDGPUTransforms/AccelerateAMDMatmul.cpp

```diff
@@ -1013,6 +1256,15 @@ class TritonAMDGPUAccelerateMatmulPass
           context, getMfmaVersion(isaFamily), matrixInstructionSize, kPack,
           /*benefit=*/2);
       break;
+    case ISAFamily::CDNA4: {
+      patterns.add<::ScaledBlockedToScaledMFMAF8F6F4>(
+          context, getMfmaVersion(isaFamily), matrixInstructionSize,
+          /*benefit=*/10);
+      patterns.add<::BlockedToMFMA, ::ScaledBlockedToMFMA>(
+          context, getMfmaVersion(isaFamily), matrixInstructionSize, kPack,
+          /*benefit=*/2);
+      break;
```

Comment:
If you move CDNA4 case to be above CDNA3 and drop these lines, then you can rely on switch-case fallthrough to avoid the duplication here.